app/src/org/odk/collect/android/activities/FormEntryActivity.java

```diff
@@ -614,16 +521,168 @@ protected void onActivityResult(int requestCode, int resultCode, Intent intent)
+     *                not attempt to scale
+     * @return a version of the image that is in its final location but has NOT been scaled, for
+     * use in displaying the captured/chosen image on the device screen when this question widget
+     * is in view (since the purpose of scaling is to limit the size of images sent to HQ, but we
+     * still want best possible quality on the device)
+     */
+    private File moveAndScaleImage(File originalImage, boolean shouldScale) {
+        // We want to save our final image file in the instance folder for this form, so that it
+        // gets sent to HQ with the form
+        String instanceFolder =
+                mInstancePath.substring(0, mInstancePath.lastIndexOf("/") + 1);
+        String imageFilename = System.currentTimeMillis() + ".jpg";
+        String finalFilePath = instanceFolder + imageFilename;
+
+        boolean savedScaledImage = false;
+        if (shouldScale) {
+            ImageWidget currentWidget = (ImageWidget)getPendingWidget();
+            int maxDimen = currentWidget.getMaxDimen();
+            if (maxDimen != -1) {
+                savedScaledImage = FileUtils.scaleImage(originalImage, finalFilePath, maxDimen);
+            }
+        }
 
-        //Ugh, copied from the odkview mostly, that's stupid
-        for(QuestionWidget q : ((ODKView)mCurrentView).getWidgets()) {
-            //Figure out if we have a pending intent widget
-            if (q instanceof IntentWidget) {
-                if(((IntentWidget) q).getFormId().equals(mFormController.getPendingCalloutFormIndex())) {
-                    bestMatch = (IntentWidget)q;
-                }
+        File fileToReturn;
+        if (!savedScaledImage) {
+            // If we didn't create a scaled image and save it to the final path, then relocate the
+            // original image from the temp filepath to our final path
+            File finalFile = new File(finalFilePath);
+            if (!originalImage.renameTo(finalFile)) {
+                fileToReturn = originalImage;
+                Log.e(TAG, "Failed to rename " + originalImage.getAbsolutePath());
+            } else {
+                fileToReturn = finalFile;
+                Log.i(TAG, "renamed " + originalImage.getAbsolutePath() + " to " + finalFile.getAbsolutePath());
+            }
+        } else {
+            // Otherwise, relocate the original image to a raw/ folder, so that we still have access
+            // to the unmodified version
+            String rawDirPath = instanceFolder + "/raw";
+            File rawDir = new File(rawDirPath);
+            if (!rawDir.exists()) {
+                rawDir.mkdir();
+            }
+            File rawImageFile = new File (rawDirPath + "/" + imageFilename);
+            if (!originalImage.renameTo(rawImageFile)) {
+                fileToReturn = originalImage;
+                Log.e(TAG, "Failed to rename " + originalImage.getAbsolutePath());
```

Comment:
This pattern is followed across this method, but the outcomes are confusing to me.

What happens if the renames/modifications don't work? Is there a reason we're handling the if/else's if we can't complete the requested action?